commands/lib/search_test.go

```diff
@@ -94,3 +94,27 @@ func TestSearchLibraryFields(t *testing.T) {
 	require.Len(t, res, 19)
 	require.Equal(t, "FlashStorage", res[0])
 }
+
+func TestSearchLibraryWithQualifiers(t *testing.T) {
+	lm := librariesmanager.NewLibraryManager(fullIndexPath, nil)
+	lm.LoadIndex()
+
+	query := func(q string) []string {
+		libs := []string{}
+		for _, lib := range searchLibrary(&rpc.LibrarySearchRequest{SearchArgs: q}, lm).Libraries {
+			libs = append(libs, lib.Name)
+		}
+		return libs
+	}
+
+	res := query("name:FlashStorage")
+	require.Len(t, res, 7)
+
+	res = query("name=FlashStorage")
+	require.Len(t, res, 1)
+	require.Equal(t, "FlashStorage", res[0])
+
+	res = query("name=\"Painless Mesh\"")
+	require.Len(t, res, 1)
+	require.Equal(t, "Painless Mesh", res[0])
```

Comment:
I would like to add some more tests regarding:
- the usage of multiple qualifiers
- multiple qualifiers and a query (`author:arduino architecture:avr led`) 
- mixing different multiple qualifiers (`author:arduino architecture=avr`)
- providing some "invalid" inputs like wrong escaping, missing closed `"` etc (example: `name=\"mesh "`

Reply:
Regarding the "invalid" inputs - since there is no mechanism to indicate syntax errors, should it try to be as permissive as possible?

A specific example is `name:"First Second` (no closing double-quote). The current code will return everything after the opening quote as a single token, meaning that this search returns the same results as `name:"First Second"`. I think that's reasonable behavior and there wouldn't be benefit in rejecting the former query as malformed.

Reply:
@zvonler yeah, it's totally fine to be permissive. I quoted "invalid" because we can handle such cases (just wanted to make it obvious on the test side :nerd_face: ). I quickly looked at the tests, and it's looking great.
I'll be traveling this weekend so you'll have to be patient till Monday:pray: